### PR TITLE
build(Windows): fix redoing version generation

### PR DESCRIPTION
**Pull request by 3N4N**
Problem: On Windows, neovim's version is generated every time nvim is built, even if code hasn't been changed.  That is because version generation is done based on a hash matching of a file and the content of the file.  And in Windows they don't match, because of the DOS line-endings.

Solution: Write the file containing nvim version with UNIX line-endings.

cc @dundargoc

### CODE REVIEW

File: cmake/GenerateVersion.cmake [hunk 1]
```diff
@@ -33,4 +33,7 @@ endif()
 if(NOT "${NVIM_VERSION_HASH}" STREQUAL "${CURRENT_VERSION_HASH}")
   message(STATUS "Using NVIM_VERSION: ${NVIM_VERSION}")
   file(WRITE "${OUTPUT}" "${NVIM_VERSION_STRING}")
+  if(WIN32)
+    configure_file("${OUTPUT}" "${OUTPUT}" NO_SOURCE_PERMISSIONS @ONLY NEWLINE_STYLE UNIX)
```
Comment (dundargoc) [inline cmake/GenerateVersion.cmake:37]: `NO_SOURCE_PERMISSIONS` is only available from cmake 3.19, so we should probably not have that. I don't think it's needed regardless?

Same with `@ONLY`, I don't think it's needed, although it doesn't hurt either.

Comment (3N4N) [reply cmake/GenerateVersion.cmake]: I removed `NO_SOURCE_PERMISSIONS` (it was supposed to be `USE_SOURCE_PERMISSIONS`; I don't know how I copy-pasted the wrong thing when it's in capital letters).  Keeping `@ONLY` unless you confirm for remove.

**Review by dundargoc — APPROVED** [review]

File: cmake/GenerateVersion.cmake (hunk 1 above)
Comment (3N4N) [reply cmake/GenerateVersion.cmake]: Done.

**Comment by dundargoc** [comment]
LGTM, just gonna wait for the CI to turn green.

**Comment by dundargoc** [comment]
Thank you :)

**Pull request merged**